Modules/Loadable/Markups/Widgets/qMRMLMarkupsInteractionHandleWidget.cxx

```diff
@@ -104,20 +133,35 @@ void qMRMLMarkupsInteractionHandleWidget::updateWidgetFromMRML()
   this->setEnabled(d->DisplayNode != nullptr);
   d->overallVisibilityCheckBox->setEnabled(d->DisplayNode != nullptr);
   d->translateVisibilityCheckBox->setEnabled(d->DisplayNode != nullptr);
+  d->translateXCheckBox->setEnabled(d->DisplayNode != nullptr);
+  d->translateYCheckBox->setEnabled(d->DisplayNode != nullptr);
+  d->translateZCheckBox->setEnabled(d->DisplayNode != nullptr);
+  d->translateViewPlaneCheckBox->setEnabled(d->DisplayNode != nullptr);
   d->rotateVisibilityCheckBox->setEnabled(d->DisplayNode != nullptr);
+  d->rotateXCheckBox->setEnabled(d->DisplayNode != nullptr);
+  d->rotateYCheckBox->setEnabled(d->DisplayNode != nullptr);
+  d->rotateZCheckBox->setEnabled(d->DisplayNode != nullptr);
   d->scaleVisibilityCheckBox->setEnabled(d->DisplayNode != nullptr);
+  d->scaleXCheckBox->setEnabled(d->DisplayNode != nullptr);
+  d->scaleYCheckBox->setEnabled(d->DisplayNode != nullptr);
+  d->scaleZCheckBox->setEnabled(d->DisplayNode != nullptr);
   d->interactionHandleScaleSlider->setEnabled(d->DisplayNode != nullptr);
+
   if (!d->DisplayNode)
     {
     return;
     }
 
-  // Scale handles currently not implemented for representations other than ROI
+  // Scale handles currently not implemented for representations other than ROI and plane
   // Disable by default.
-  if (!vtkMRMLMarkupsROIDisplayNode::SafeDownCast(d->DisplayNode))
+  if (!vtkMRMLMarkupsROIDisplayNode::SafeDownCast(d->DisplayNode) && !vtkMRMLMarkupsPlaneDisplayNode::SafeDownCast(d->DisplayNode))
     {
     // Scale handles not currently implemented for non ROI nodes
     d->scaleVisibilityCheckBox->setVisible(false);
+    d->scaleXCheckBox->setVisible(false);
```

Comment:
It looks a bit strange that the labels remain there but the checkboxes disappear. Instead of making them disappear could you uncheck and disable the options that are not applicable to the node type? (we could hide the labels as well, but then the layout would jump around when browsing the tree, so it is probably better to keep displaying all rows)